### ISSUE TITLE
Automated PDF Testing

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#199]
## Problem
No existing tool runs automated tests against multiple PDFs with thousands of pages.

## Solution / Goal
Build a web service to scan multiple PDF documents against tests that are defined in the Matterhorn and WCAG documents. The web service would provide alerts to Equalify. 

## User Experience
In Equalify, under "Scan Profiles", a user would add a URL and select a "PDF Crawl" Scan Type. The URL could include one or multiple PDFs. 

When a user runs a scan, Equalify sends the URL to the integration's web service. That service searches the web page for PDFs. Automated tests are run against every PDF that is found. Alerts are returned with messages that allow for quick remediation.

## Development
We'll build a web service around the [VeraPDF CLI](https://docs.verapdf.org/cli/) and [related webservice](https://github.com/veraPDF/veraPDF-rest). 

## Additional Notes
- Axe published [a PDF ruleset](https://dequeuniversity.com/rules/axe/pdf/2.0/).
- The integration could open a PDF in Acrobat, run an accessibility scan, then translate the [HTML output](https://github.com/bbertucc/equalify/files/11042342/homeacrordrunified18_2018.pdf.accreport.html.zip) as JSON returned to Equalify.
- Tested [VeraPDF](https://verapdf.org/), [PDfix](https://pdfix.net/), and [PAC 2021](https://pdfua.foundation/en/pdf-accessibility-checker-pac/) as other options - all those options are not as simple as [Acrobat accessiblity](https://helpx.adobe.com/acrobat/using/create-verify-pdf-accessibility.html). (For instance, PDfix requires several clicks to create a UA PDF where Acrobat does that with an "Autotag Document" button.) That said, Acrobat misses content errors that the others find.  
- After we figure out how to test, we can remediate. For instance, the same automation that we used for testing with Acrobat accessibility could add data like language. We could also use a service like [iText](https://kb.itextpdf.com/home/it7kb/ebooks/itext-7-jump-start-tutorial-for-java/chapter-7-creating-pdf-ua-and-pdf-a-documents).
- This tool looks promising, though I have not gotten it working on Mac OS: https://github.com/accessibility-luxembourg/simplA11yPDFCrawler
- The "Matterhorn Protocol" includes automated and manual testing: https://www.pdfa.org/resource/the-matterhorn-protocol/
- PAC 2021 is introduced in this lecture: https://link.springer.com/chapter/10.1007/978-3-319-08596-8_31

**Comment by bbertucc** [comment]
Closing this issue for lack of activity, but feel free to open the issue with more examples of how this feature would help.
